### PR TITLE
Setup integration test infrastructure

### LINKED ISSUE #3: Problem: no integration tests framework
We can do some specific integration test for cronos in this repo, need an integration test framework to do that, maybe migrate the one used in chain-main repository.

### PR DESCRIPTION
**Pull request by yihuang**
Closes #3

- use nix
- support both cronos and geth

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: nix/scripts.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      pwdfile=$(mktemp /tmp/password.XXXXXX)
+      tmpfile=$(mktemp /tmp/validator-key.XXXXXX)
+
+      cat > $pwdfile << EOF
+      123456
+      EOF
+
+      # import validator key
+      cat > $tmpfile << EOF
+      826E479F5385C8C32CD96B0C0ACCDB8CC4FA5CACCC1BE54C1E3AA4D676A6EFF5
```
Comment (tomtau) [inline nix/scripts.nix:34]: from Geth genesis?

**Review by tomtau — APPROVED** [review]
looks all right, just not sure of some of the hardcoded address constants where they come from

File: nix/scripts.nix (hunk 1 above)
Comment (yihuang) [reply nix/scripts.nix:34]: there's no private key in the genesis file though.

File: integration_tests/utils.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+import eth_account
+import eth_utils
+import rlp
+import yaml
+from dateutil.parser import isoparse
+from pystarport import cluster, ledger
+from pystarport.ports import rpc_port
+
+KEYS = {
+    "validator": "826E479F5385C8C32CD96B0C0ACCDB8CC4FA5CACCC1BE54C1E3AA4D676A6EFF5",
```
[the remark this answers is not in the document]
Comment (yihuang) [reply integration_tests/utils.py:20]: yeah, it's possible, need to derive it on the fly though, based on the mnemonic words in the config file.

File: default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{ system ? builtins.currentSystem, pkgs ? import ./nix { inherit system; } }:
+pkgs.buildGoApplication rec {
+  pname = "cronosd";
+  version = "0.0.0";
```
[the remark this answers is not in the document]
Comment (yihuang) [reply default.nix:4]: done

File: integration_tests/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (yihuang) [reply integration_tests/utils.py:20]: I'd like to fix the hardcoded stuff in another PR.
https://github.com/crypto-org-chain/cronos/issues/10

File: nix/scripts.nix (hunk 1 above)
Comment (yihuang) [reply nix/scripts.nix:34]: I'd like to fix the hardcoded stuff in another PR.
https://github.com/crypto-org-chain/cronos/issues/10

**Comment by yihuang** [comment]
Will run integration tests in CI in another PR.

**Pull request merged**